python/src/opendp/mod.py

```diff
@@ -1293,24 +1293,8 @@ def exponential_bounds_search(
     """
 
     # try to infer T
-    if T is None:
-        def _check_type(v):
-            try:
-                predicate(v)
-            except TypeError:
-                return False
-            except OpenDPException as e:
-                if "No match for concrete type" in (e.message or ""):
-                    return False # pragma: no cover
-            return True
-        
-        if _check_type(0.):
-            T = float
-        elif _check_type(0):
-            T = int
-        else:
-            raise TypeError("unable to infer type `T`; pass the type `T` or bounds")  # pragma: no cover
-
+    T = T or _infer_type(predicate)
```

Comment:
Perhaps
```suggestion
    T = T if T is not None else _infer_type(predicate)
```
depending on the behavior you want for `0`, `False`, or `""`? But keep it simple if it doesn't matter.

Reply:
If `T` is falsey, then it needs to be inferred.